janusgraph-core/src/main/java/org/janusgraph/core/attribute/Geo.java

```diff
@@ -108,6 +108,34 @@ public boolean hasNegation() {
         public JanusGraphPredicate negate() {
             throw new UnsupportedOperationException();
         }
+    },
+
+    /**
+     * Whether one geographic region completely contains another
+     */
+    CONTAINS {
+        @Override
+        public boolean test(Object value, Object condition) {
+            Preconditions.checkArgument(condition instanceof Geoshape);
+            if (value == null) return false;
+            Preconditions.checkArgument(value instanceof Geoshape);
+            return ((Geoshape) value).contains((Geoshape) condition);
+        }
+
+        @Override
+        public String toString() {
+            return "contains";
+        }
+
+        @Override
+        public boolean hasNegation() {
+            return false;
```

Comment:
A JanusGraphPredicate can only have a negation if it can be defined directly as another JanusGraphPredicate. This isn't possible for Geo.Contains (e.g. negation is not Geo.WITHIN for example).

Note this is a separate question from whether the negation of the contains Contains spatial predicate exists mathematically, though I'm not sure it does because of subtleties in the definition.

https://en.wikipedia.org/wiki/DE-9IM#Spatial_predicates
http://lin-ear-th-inking.blogspot.com/2007/06/subtleties-of-ogc-covers-spatial.html